### PR TITLE
Ignore --ignore-fee-rates on mutual close, don't massively overpay with LDK nodes

### PR DESCRIPTION
**Pull request by rustyrussell**
https://github.com/ElementsProject/lightning/issues/7242 describes the pain this causes :(

Fixes https://github.com/ElementsProject/lightning/issues/7242

### CODE REVIEW

**Comment by TonyGiorgio** [comment]
Concept ACK, appreciate this. Have not tested this yet but from investigations on our side, it sounds like this would fix the issue.

**Review by vincenzopalazzo — APPROVED** [review]
ACK 0567ac4932c0fb02a778a8dbf76df54ad97087e2

Make sense, iirc we use this ignore fee to avoid disable the channel but when we need to touch the chain it is good that we enforce our point of view, this should also possible bugs

Thanks

File: tests/test_closing.py [hunk 1]
```diff
@@ -4042,3 +4042,19 @@ def test_closing_no_anysegwit_retry(node_factory, bitcoind):
...
+                                               'ignore-fee-limits': True},
+                                              {'may_reconnect': True}])
+
+    # l2's feerates go up.  A lot!
+    l2.set_feerates((100000, 100000, 100000, 100000))
+    l2.restart()
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+
+    # This fails to negotiate.
+    fut = executor.submit(l1.rpc.close, l2.info['id'])
```
Comment (vincenzopalazzo) [inline tests/test_closing.py:4059]:
```suggestion
    _ = executor.submit(l1.rpc.close, l2.info['id'])
```

I think we need just to fix the CI

```
cc common/utils.c
ld tools/check-bolt
tests/test_closing.py:4059:5: F841 local variable 'fut' is assigned to but never used
make: *** [Makefile:535: check-python-flake8] Error 1
rm external/build-x86_64-linux-gnu/libwally-core-build/src/secp256k1/libsecp256k1.la
```

Comment (rustyrussell) [reply tests/test_closing.py]: Ah, I can just omit the assignment completely, thanks!

**Comment by vincenzopalazzo** [comment]
Ok the CI cached the following crash, that looks like unrelated, but I will give another un before merging the PR

```
lightningd-1 2024-04-23T08:08:42.059Z DEBUG   lightningd: Adding block 101: 560466b8148c0f2f6cab1910d1d8eecd043617ccba5cbf94bf0312d56aac271e
lightningd-1 2024-04-23T08:08:42.721Z DEBUG   gossipd: REPLY WIRE_GOSSIPD_NEW_BLOCKHEIGHT_REPLY with 0 fds
lightningd-1 2024-04-23T08:08:42.878Z INFO    plugin-pay: Killing plugin: exited during normal operation
lightningd-1 2024-04-23T08:08:42.878Z **BROKEN** plugin-pay: Plugin marked as important, shutting down lightningd!
```

**Review by niftynei — APPROVED** [review]
ACK fc45c1f2bebbeb92a4c802c178d58cddcf5b5a59

**Comment by rustyrussell** [comment]
Fixed timeout caused by close taking forever in test_fee_limits...

**Review by vincenzopalazzo — COMMENTED** [review]
```
genidx doc/index.rst
Checking for generated files being changed by make
git diff --exit-code HEAD
diff --git a/contrib/pyln-grpc-proto/pyln/grpc/node_pb2.py b/contrib/pyln-grpc-proto/pyln/grpc/node_pb2.py
index 2ea52d86d..665aa592c 100644
--- a/contrib/pyln-grpc-proto/pyln/grpc/node_pb2.py
+++ b/contrib/pyln-grpc-proto/pyln/grpc/node_pb2.py
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
 # source: node.proto
+# Protobuf Python Version: 4.25.1
 """Generated protocol buffer code."""
 from google.protobuf import descriptor as _descriptor
 from google.protobuf import descriptor_pool as _descriptor_pool
diff --git a/contrib/pyln-grpc-proto/pyln/grpc/primitives_pb2.py b/contrib/pyln-grpc-proto/pyln/grpc/primitives_pb2.py
index b55614db6..955eb91af 100644
--- a/contrib/pyln-grpc-proto/pyln/grpc/primitives_pb2.py
+++ b/contrib/pyln-grpc-proto/pyln/grpc/primitives_pb2.py
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
 # source: primitives.proto
+# Protobuf Python Version: 4.25.1
 """Generated protocol buffer code."""
 from google.protobuf import descriptor as _descriptor
 from google.protobuf import descriptor_pool as _descriptor_pool
make: *** [Makefile:599: check-gen-updated] Error 1
```

Not clear why this is happening, but looks like the diff for the pylon-grpc-proto package is not necessary.

**Review by vincenzopalazzo — APPROVED** [review]
ACK a86e31a80e67c9f3bc66d89245e2c0b9356f4796

**Pull request merged**